### PR TITLE
Adjust period after each fh.update() call

### PR DESCRIPTION
**Pull request by tedyu**
In realFsHandler#trackUsage, the error return from the first call to fh.update() is ignored.

This PR extracts the period adjustment logic into a func and calls this func in place of original fh.update()

### CODE REVIEW

**Comment by tedyu** [comment]
@dashpole 
Tests passed.

File: container/common/fsHandler.go [hunk 1]
```diff
@@ -106,23 +106,26 @@ func (fh *realFsHandler) update() error {
...
-			if err := fh.update(); err != nil {
-				klog.Errorf("failed to collect filesystem stats - %v", err)
-				fh.period = fh.period * 2
-				if fh.period > maxBackoffFactor*fh.minPeriod {
-					fh.period = maxBackoffFactor * fh.minPeriod
-				}
-			} else {
-				fh.period = fh.minPeriod
-			}
+			periodAdjuster()
```
Comment (dashpole) [inline container/common/fsHandler.go]: Can we instead just move this logic to the top of the loop?  I.E.
```golang
for {
    ... all of the logic ...
    select {
    case <-fh.stopChan:
        return
    case <-time.After(fh.period):
    }
}
```

Also, lets include the logging for long operations in there as well.

Comment (tedyu) [reply container/common/fsHandler.go]: @dashpole 
Done.

File: container/common/fsHandler.go [hunk 2]
```diff
@@ -106,23 +106,22 @@ func (fh *realFsHandler) update() error {
...
-			if err := fh.update(); err != nil {
-				klog.Errorf("failed to collect filesystem stats - %v", err)
-				fh.period = fh.period * 2
-				if fh.period > maxBackoffFactor*fh.minPeriod {
-					fh.period = maxBackoffFactor * fh.minPeriod
-				}
-			} else {
-				fh.period = fh.minPeriod
-			}
 			duration := time.Since(start)
```
Comment (dashpole) [inline container/common/fsHandler.go]: please move the start, and duration logging here up as well

File: container/common/fsHandler.go [hunk 3]
```diff
@@ -106,31 +106,30 @@ func (fh *realFsHandler) update() error {
...
+		if err := fh.update(); err != nil {
+			klog.Errorf("failed to collect filesystem stats - %v", err)
+			fh.period = fh.period * 2
+			if fh.period > maxBackoffFactor*fh.minPeriod {
+				fh.period = maxBackoffFactor * fh.minPeriod
+			}
+		} else {
+			fh.period = fh.minPeriod
+		}
+		start := time.Now()
```
Comment (dashpole) [inline container/common/fsHandler.go]: start needs to be defined before the update.  The current change makes duration ~= 0 always.

Comment (tedyu) [reply container/common/fsHandler.go]: @dashpole 
Corrected.

**Pull request merged**